src/components/ToolbarMenu.js

```diff
@@ -14,27 +14,52 @@ import React, { PropTypes } from 'react';
  *
  * A menu style button to use within the Toolbar.
  */
-export function ToolbarMenu({ title, label, children }) {
-  return (
-    <a
-      className="toolbar-menu toolbar-button"
-      onMouseDown={preventDefault}
-      title={title}>
-      {label}
-      <svg width="14" height="8">
-        <path fill="#666" d="M 5 1.5 L 14 1.5 L 9.5 7 z" />
-      </svg>
-      <ul className="toolbar-menu-items">
-        {children}
-      </ul>
-    </a>
-  );
-}
+export class ToolbarMenu extends React.Component {
+  static propTypes = {
+    title: PropTypes.string,
+    label: PropTypes.string,
+  }
 
-ToolbarMenu.propTypes = {
-  title: PropTypes.string,
-  label: PropTypes.string,
-};
+  constructor(props) {
+    super(props);
+    this.state = { visibility: 'hidden' };
+  }
+
+  render() {
+    const ulStyle = {
+      visibility: this.state.visibility
+    };
+    return (
+      <a
+        className="toolbar-menu toolbar-button"
+        onClick={this.handleOpen.bind(this)}
+        onMouseDown={preventDefault}
+        ref={node => {this._node = node;}}
+        title={this.props.title}>
+        {this.props.label}
+        <svg width="14" height="8">
+          <path fill="#666" d="M 5 1.5 L 14 1.5 L 9.5 7 z" />
+        </svg>
+        <ul className="toolbar-menu-items" style={ulStyle}>
+          {this.props.children}
+        </ul>
+      </a>
+    );
+  }
+
+  handleClick(e) {
+    if (this._node !== e.target) {
+      e.preventDefault();
+      this.setState({ visibility: 'hidden' });
+    }
+  }
+
+  handleOpen = e => {
+    e.preventDefault();
+    this.setState({ visibility: 'visible' });
+    document.addEventListener('click', this.handleClick.bind(this));
```

Comment:
For hygiene, we should make sure we remove this event listener on unmount or close. The currently implementation will re-bind repeatedly.